config/routes.rb

```diff
@@ -23,11 +25,15 @@
     # Newer implementation: will supersede the above.
     get "users/two_factor_authentication/setup", to: "users/two_factor_authentication/setup#show"
     put "users/two_factor_authentication/setup", to: "users/two_factor_authentication/setup#update"
-    get "users/two_factor_authentication/email", to: "users/two_factor_authentication/email#show"
-    post "users/two_factor_authentication/email", to: "users/two_factor_authentication/email#update"
+
+    get "users/two_factor_authentication/email/resend", to: "users/two_factor_authentication/email#resend"
```

Comment:
I would have preferred to keep the URLs restful as much as possible, and annotate actions as parameters rather than entities on their own. So, since `resend` is an action on the `email` entity, my preference would have been to specify it as a parameter on the `email` resource (e.g. `/email?action=resend` or `/email?resend=true`, etc.).

I feel the same about the other routes below as well, with paths ending with an action or which are there just for convenience rather than leading up to an entity.

Reply:
Maybe I can rename the routes to 'email/confirmation' and 'email/resend_form' or something so they are nouns? That would make it Restful again. I'd rather not add routing complexity in the controller. What do you think?

Reply:
Actually, having looked at the behaviour on the browser, I'm happy with `..two_factor_authentication/email/resend` – apologies for the confusion.